storage/storage.c

```diff
@@ -1175,6 +1331,36 @@ secbool storage_change_pin(uint32_t oldpin, uint32_t newpin,
   return ret;
 }
 
+secbool storage_has_wipe_code(void) {
+  if (sectrue != initialized || sectrue != unlocked) {
+    return secfalse;
+  }
+
+  return is_not_wipe_code(WIPE_CODE_EMPTY);
+}
+
+secbool storage_change_wipe_code(uint32_t pin, const uint8_t *ext_salt,
+                                 uint32_t wipe_code) {
+  if (sectrue != initialized || (pin != PIN_EMPTY && pin == wipe_code)) {
+    memzero(&pin, sizeof(pin));
+    memzero(&wipe_code, sizeof(wipe_code));
+    return secfalse;
+  }
+
+  ui_total = DERIVE_SECS;
+  ui_rem = ui_total;
+  ui_message = (pin != PIN_EMPTY && wipe_code == PIN_EMPTY) ? VERIFYING_PIN_MSG
+                                                            : PROCESSING_MSG;
+
+  secbool ret = secfalse;
+  if (sectrue == unlock(pin, ext_salt)) {
+    ret = set_wipe_code(wipe_code);
+  }
+  memzero(&pin, sizeof(pin));
+  memzero(&wipe_code, sizeof(wipe_code));
+  return ret;
+}
+
 void storage_wipe(void) {
```

Comment:
You didn't implement this feature:
> When the wipe is triggered, the **EDEK value in the storage will be immediately overwritten with zero bytes**, then norcow_wipe() will be called.

Reply:
This is marked as TODO somewhere. Requires further research, a separate issue created.